src/runtime_src/core/tools/xbmgmt2/flash/xospiversal.cpp

```diff
@@ -36,10 +35,18 @@ int XOSPIVER_Flasher::xclUpgradeFirmware(std::istream& binStream)
 
     std::cout << "INFO: ***PDI has " << total_size << " bytes" << std::endl;
 
+    auto fd = m_device->file_open("ospi_versal", O_RDWR); 

```

Comment:
Should file_open throw on error?  And then try/catch printing the error as you do here.  At least then the error cannot be ignored.